src/orion/core/utils/random_state.py

```diff
@@ -77,3 +78,26 @@ def seed(cls, base_seed: int | None) -> RandomState:
             # Add the base_seed property in this case.
             random_state = replace(random_state, base_seed=base_seed)
         return random_state
+
+
+@contextmanager
+def control_randomness(random_state: RandomState | None):
+    """Seeds the randomness inside the indented block of code using `self.random_state`.
+
+    NOTE: This only has an effect if `seed_rng` was called previously, i.e. if
+    `self.random_state` is not None.
+    """
+    if random_state is None:
```

Comment:
Could probably remove this case, and mark the argument as required instead of optional.
It doesn't really make sense to call `control_randomness()` and for it to not do anything.

Reply:
If the algo has no seed. We could have the if-else out-side control-randomness though.

Reply:
Related discussion: https://github.com/Epistimio/orion/pull/903#discussion_r931059989